vimdoc/docline.py

```diff
@@ -180,16 +180,29 @@ def Update(self, block):
 class Section(BlockDirective):
   REGEX = regex.section_args
 
-  def Assign(self, name, ident):
+  def __init__(self, args):
+    # Used to collect children at Module processing time.
+    self.parent_id = None
+    super(Section, self).__init__(args)
+
+  def Assign(self, name, ident, parent):
     self.name = name.replace('\\,', ',').replace('\\\\', '\\')
+
     if ident is None:
       # If omitted, it's the name in lowercase, with spaces converted to dashes.
       ident = self.name.lower().replace(' ', '-')
     self.id = ident
 
+    # This is potentially a problem if the parent layer name used spaces and an
+    # id was autogenerated; the parent_id needs to match the hyphenated form.
+    # We could enforce an explicit id field for sections that are used as a
```

Comment:
realised this was not possible because we don't know if the parent has been read at the time we read the child, and when we assemble the tree we don't know if the id was explicit or autogenerated. i added a hint in the error message instead.